tests/discovery.js

```diff
@@ -1,439 +1,461 @@
-    })
-  }
-
-  let count = 0
-  async function step() {
-    count++
-    if (count === connections) {
-      await Promise.all(
-        instances.map(async (instance) => {
-          await instance.discovery.leave(core1.discoveryKey)
-          await instance.discovery.destroy()
-        })
-      )
-      await testnet.destroy()
-    }
-  }
-
-  for (const instance of instances) {
-    await instance.discovery.join(core1.discoveryKey)
-  }
-})
+import { randomBytes } from 'node:crypto'
+import { KeyManager } from '@mapeo/crypto'
 
 test('discovery - mdns', async (t) => {
-  t.plan(2)
-
-  const keyPair = createCoreKeyPair('mdns-peer-discovery')
-
-  const identity1 = createIdentityKeys()
-  const identityPublicKey1 = identity1.identityKeyPair.publicKey.toString('hex')
-
-  const identity2 = createIdentityKeys()
-  const identityPublicKey2 = identity2.identityKeyPair.publicKey.toString('hex')
-
-  const discover1 = new Discovery({
-    identityKeyPair: identity1.identityKeyPair,
-    mdns: true,
-    dht: false,
-  })
-
-  const discover2 = new Discovery({
-    identityKeyPair: identity2.identityKeyPair,
-    mdns: true,
-    dht: false,
-  })
-
-  await discover1.ready()
-  await discover2.ready()
-
-  discover1.on('connection', async (connection, peer) => {
-    t.ok(peer.identityPublicKey === identityPublicKey2, 'match key of 2nd peer')
-    await step()
-  })
-
-  discover2.on('connection', async (connection, peer) => {
-    t.ok(peer.identityPublicKey === identityPublicKey1, 'match key of 1st peer')
-    await step()
-  })
-
-  let count = 0
-  async function step() {
-    count++
-    if (count === 2) {
-      await discover1.leave(keyPair.publicKey)
-      await discover2.leave(keyPair.publicKey)
-      await discover1.destroy()
-      await discover2.destroy()
-    }
-  }
-
-  await discover1.join(keyPair.publicKey)
-  await discover2.join(keyPair.publicKey)
-})
-
-test('replication - mdns', async (t) => {
-  t.plan(3)
-
-  const core1 = new Hypercore(ram)
-  await core1.ready()
-
-  const core2 = new Hypercore(ram, core1.key)
-  await core2.ready()
+  const identityKeypair1 = new KeyManager(randomBytes(16)).getIdentityKeypair()
+  const identityKeypair2 = new KeyManager(randomBytes(16)).getIdentityKeypair()
 
-  core1.append(['a', 'b', 'c'])
-
-  const identity1 = createIdentityKeys()
-  const identityPublicKey1 = identity1.identityKeyPair.publicKey.toString('hex')
-
-  const identity2 = createIdentityKeys()
-  const identityPublicKey2 = identity2.identityKeyPair.publicKey.toString('hex')
-
-  const discover1 = new Discovery({
-    identityKeyPair: identity1.identityKeyPair,
-    mdns: true,
-    dht: false,
-  })
-
-  const discover2 = new Discovery({
-    identityKeyPair: identity2.identityKeyPair,
-    mdns: true,
-    dht: false,
+  const mdnsDiscovery1 = new MdnsDiscovery({identityKeypair:identityKeypair1})
+  mdnsDiscovery1.on('connection', (stream) => {
+    console.log('found peer 2')
   })
+  await mdnsDiscovery1.start()
```

Comment:
call `start()` on both discovery instances at the same time (e.g. don't await the first) so that we can catch potential duplicate connections.